rkt/gc.go

```diff
@@ -188,50 +235,13 @@ func deletePod(p *pod) {
 	}
 
 	if p.isExitedGarbage {
-		s, err := store.NewStore(globalFlags.Dir)
-		if err != nil {
-			stderr("Cannot open store: %v", err)
-			return
-		}
-		defer s.Close()
-		stage1TreeStoreID, err := p.getStage1TreeStoreID()
-		if err != nil {
-			stderr("Error getting stage1 treeStoreID: %v", err)
-			return
-		}
-		stage1RootFS := s.GetTreeStoreRootFS(stage1TreeStoreID)
-
-		// execute stage1's GC
-		if err := stage0.GC(p.path(), p.uuid, stage1RootFS, globalFlags.Debug); err != nil {
-			stderr("Stage1 GC of pod %q failed: %v", p.uuid, err)
-			return
-		}
-
-		if p.usesOverlay() {
-			apps, err := p.getApps()
-			if err != nil {
-				stderr("Error retrieving app hashes from pod manifest: %v", err)
-				return
+		if err := cleanExitedGarbage(p); err != nil {
```

Comment:
I agree with the example you've given and have moved that out of the checkExitGarbage function. However, I don't think there is a need to get more fancy with the rest of the failures. If the next one we get to fails (treestoreid), or any thereafter, this means the pod directory is in a weird state. IMHO we should just bail and force clean at that point.